scalableminds/webknossos
# Better DOM snapshot diffs

This PR should make it a lot easier to understand differences in the DOM snapshots generated in our snapshot end-to-end tests. Previously, removing a Spinner caused all elements that were wrapped by that spinner to display as changed, due to the changed indentation level (and changed JSON structure of the snapshot).
Now, the snapshots are saved as multiline HTML strings (which reduces their size from ~58000 lines to ~7000 lines) and, additionally, leading whitespace is removed. The result is that all lines have the same indentation level, but removing said Spinner only causes two line changes (as the indentation level of the wrapped elements is disregarded). It is harder to see the DOM structure in the diff now, but the advantage of having much simpler diffs is worth this imo. One is still able to look at the debug html files (in the `debug-htmls` folder), that contain the indented original DOM structure and which are written to disk during testing as well.

### Steps to test:
- Successful CI run
- Check out the branch locally, remove a spinner and have a look at the diff result. Unfortunately a lot of unchanged lines are printed by ava as well, but I couldn't find an option to disable that.

------
